### PR TITLE
Update bookshelf: tag to current user choice for book (BL-10031)

### PR DESCRIPTION
**Pull request by StephenMcConnel**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloom-harvester/164)

### CODE REVIEW

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 3 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloom-harvester/164)* status: 3 of 4 files reviewed, 1 unresolved discussion (waiting on @andrew-polk and @StephenMcConnel)

___
*[src/Harvester/Harvester.cs, line 660 at r1](https://reviewable.io/reviews/bloombooks/bloom-harvester/164#-MwmEWvsDv0vUBWt7eHe:-MwmEWvsDv0vUBWt7eHf:b-tkhx7r) ([raw file](https://github.com/bloombooks/bloom-harvester/blob/870c41630ba425f604829d5240f6870ca0a0c146/src/Harvester/Harvester.cs#L660)):*
> ```Smalltalk
> 				_currentBookId = null;
> 				book.Model.FlushUpdateToDatabase(_parseClient, _options.ReadOnly);
> 				if (!_options.ReadOnly)
> ```

Can this be

if (!_options.ReadOnly)
	book.Model.UpdateBookshelfTag(book.Analyzer.GetBookshelf(), _parseClient);
book.Model.FlushUpdateToDatabase(_parseClient, _options.ReadOnly);

and remove the flush from UpdateBookshelfTag?
				
In other words, do we need to flush and then maybe flush? Or can we just flush once?



_Code quote:_
```Smalltalk
				book.Model.FlushUpdateToDatabase(_parseClient, _options.ReadOnly);
				if (!_options.ReadOnly)
					book.Model.UpdateBookshelfTag(book.Analyzer.GetBookshelf(), _parseClient);
```

**Review by StephenMcConnel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloom-harvester/164)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @andrew-polk)

___
*[src/Harvester/Harvester.cs, line 660 at r1](https://reviewable.io/reviews/bloombooks/bloom-harvester/164#-MwmEWvsDv0vUBWt7eHe:-MwmrWFC3W9gJzCMaV1y:b-896fix) ([raw file](https://github.com/bloombooks/bloom-harvester/blob/870c41630ba425f604829d5240f6870ca0a0c146/src/Harvester/Harvester.cs#L660)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Can this be

if (!_options.ReadOnly)
	book.Model.UpdateBookshelfTag(book.Analyzer.GetBookshelf(), _parseClient);
book.Model.FlushUpdateToDatabase(_parseClient, _options.ReadOnly);

and remove the flush from UpdateBookshelfTag?
				
In other words, do we need to flush and then maybe flush? Or can we just flush once?
</blockquote></details>

Done.

**Review by andrew-polk — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloom-harvester/164)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @StephenMcConnel)
